cfme/containers/image.py

```diff
@@ -59,6 +59,44 @@ def perform_smartstate_analysis(self, wait_for_finish=False):
                 raise TimedOutError('Timeout exceeded, Waited too much time for SSA to finish ({}).'
                                     .format(ssa_timeout))
 
+    def check_compliance(self, timeout=240):
+        """Initiates compliance check and waits for it to finish."""
+        navigate_to(self, 'Details')
+        original_state = self.compliance_status
+        tb.select('Policy',
+                  "Check Compliance of Last Known Configuration",
+                  invokes_alert=True)
+        sel.handle_alert()
+        flash.assert_no_errors()
+        wait_for(
+            lambda: self.compliance_status != original_state,
+            num_sec=timeout, delay=5, fail_func=sel.refresh,
+            message="compliance of {} checked".format(self.name)
```

Comment:
Let's tweak the behavior of this wait_for a bit.

The call to self.compliance_status is calling self.load_details, as you've stated below. This means that there's no need for the fail_func refresh parameter here as its already going to load on each execution of the func.  5 seconds is also potentially cutting it a little close for depending on nested navigation calls. Consider bumping to 10.

I'd also like to see the `compliance_status` and `original_state` included in the log message, but this poses a challenge as we don't have the current value of `compliance_status` stored anywhere that we can access it, and calling it again from `self` will force another navigation and won't give the value that was just compared.

What do you think about changing the message to at least include the original_state, and to be more descriptive of what its doing.  Consider 'compliance state of {} still matches {}'.format(self.name, original_state)